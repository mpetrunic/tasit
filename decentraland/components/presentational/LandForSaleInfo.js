import React from "react";
import { StyleSheet, View, Text, Alert } from "react-native";
import PropTypes from "prop-types";
import {
  responsiveWidth,
  responsiveFontSize,
} from "react-native-responsive-dimensions";
import Colors from "@constants/Colors";
import { Button, Icon } from "native-base";
import { formatNumber } from "../../helpers";

const onPriceInfo = () => {
  const title = "";
  const message =
    "MANA is Decentralandâ€™s fungible, ERC20 cryptocurrency token. MANA is burned or spent in exchange for LAND parcels. ";
  Alert.alert(title, message, [{ text: "Okay" }]);
};

export default function LandForSale({ landForSale }) {
  const { asset, priceMana } = landForSale;
  let { name } = asset;
  const price = formatNumber(priceMana);
  if (!name) name = "Land without name";

  // Note: Conversion to USD will be implemented on v0.2.0
  // <Text>{priceMana} MANA (~${landForSale.priceUSD})</Text>
  return (
    <View style={styles.landInfoContainer}>
      <View style={styles.landNameContainer}>
        <Text style={styles.landName}>{name}</Text>
      </View>
      <View style={styles.landPriceContainer}>
        <View>
          <Button small transparent onPress={onPriceInfo}>
            <Icon
              name="information-circle-outline"
              style={styles.landPriceIcon}
            />
          </Button>
        </View>
        <View>
          <Text style={styles.landPrice}>{price}</Text>
        </View>
      </View>
    </View>
  );
}

LandForSale.propTypes = {
  landForSale: PropTypes.object.isRequired,
};

const styles = StyleSheet.create({
  landInfoContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
  },
  landNameContainer: {
    width: responsiveWidth(50),
  },
  landName: {
    color: Colors.assetInfoText,
  },
  landPriceContainer: {
    flexDirection: "row",
    alignItems: "center",
  },
  landPriceIcon: {
    marginLeft: responsiveWidth(1.5),
    marginRight: responsiveWidth(1.5),
  },
  landPrice: {
    fontSize: responsiveFontSize(4),
    fontWeight: "bold",
    color: Colors.assetInfoText,
  },
});
