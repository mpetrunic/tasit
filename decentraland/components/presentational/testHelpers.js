import AssetTypes from "@constants/AssetTypes";
const { ESTATE, PARCEL } = AssetTypes;

export const parcel = {
  type: PARCEL,
  id: "0123456789",
  name: "Sample Parcel",
  img: "https://api.decentraland.org/v1/parcels/0/0/map.png",
};

export const estate = {
  type: ESTATE,
  id: "123",
  name: "Sample Estate",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export const estateForSale = {
  id: "1",
  priceMana: 0,
  priceUSD: 0,
  asset: estate,
};

export const parcelForSale = {
  id: "2",
  priceMana: 0,
  priceUSD: 0,
  asset: parcel,
};

export default {
  estateForSale,
  parcelForSale,
  parcel,
  estate,
};
