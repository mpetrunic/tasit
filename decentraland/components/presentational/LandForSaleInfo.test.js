import React from "react";
import { shallow } from "enzyme";
import { LandForSaleInfo, LandForSalePrice } from "./LandForSaleInfo";
import { estateForSale } from "@helpers/testHelpers";

describe("LandForSaleInfo", () => {
  it("renders the component", async () => {
    expect(
      shallow(<LandForSaleInfo landForSale={estateForSale} />)
    ).toMatchSnapshot();
  });

  describe("LandForSalePrice", () => {
    it("renders the component", async () => {
      expect(
        shallow(<LandForSalePrice landForSale={estateForSale} />)
      ).toMatchSnapshot();
    });
  });
});
