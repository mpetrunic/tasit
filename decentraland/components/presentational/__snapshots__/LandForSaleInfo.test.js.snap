// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandForSaleInfo renders the component 1`] = `
<Component
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <Component
    style={
      Object {
        "paddingTop": 13.34,
        "width": 375,
      }
    }
  >
    <Component
      style={
        Object {
          "color": "#2B2B2B",
          "fontWeight": "bold",
        }
      }
    >
      Sample Estate
    </Component>
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Component>
      <Styled(Button)
        onPress={[Function]}
        small={true}
        transparent={true}
      >
        <Styled(Icon)
          name="information-circle-outline"
          style={
            Object {
              "marginLeft": 11.25,
              "marginRight": 11.25,
            }
          }
        />
      </Styled(Button)>
    </Component>
    <Component>
      <Component
        style={
          Object {
            "color": "#2B2B2B",
            "fontSize": 61.191865835619396,
            "fontWeight": "bold",
            "paddingTop": 2.668,
          }
        }
      >
        0
      </Component>
    </Component>
  </Component>
</Component>
`;
